src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -221,6 +230,56 @@ void EventPipe::EnableViaEnvironmentVariables()
         );
         EventPipe::StartStreaming(sessionID);
     }
+#ifndef GEN_ANALYSIS_STRESS
+    if (gcGenAnalysisConfigured == GcGenAnalysisState::Uninitialized)
+    {
+        if (CLRConfig::IsConfigOptionSpecified(W("GCGenAnalysisGen")))
+        {
+            gcGenAnalysisGen = CLRConfig::GetConfigValue(CLRConfig::INTERNAL_GCGenAnalysisGen);
+            if (CLRConfig::IsConfigOptionSpecified(W("GCGenAnalysisBytes")))
+            {
+                gcGenAnalysisBytes = CLRConfig::GetConfigValue(CLRConfig::INTERNAL_GCGenAnalysisBytes);
+                gcGenAnalysisConfigured = GcGenAnalysisState::Enabled;
+            }
+            else
+            {
+                gcGenAnalysisConfigured = GcGenAnalysisState::Disabled;
+            }
+        }
+    }
+    if (gcGenAnalysisConfigured == 1 && gcGenAnalysisState == 0)
+#endif
+    {
+        LPCWSTR outputPath = nullptr;
+        outputPath = W("gcgenaware.nettrace");
+        NewHolder<EventPipeProviderConfiguration> pProviders = nullptr;
+        int providerCnt = 1;
+        pProviders = new EventPipeProviderConfiguration[providerCnt];
+        const uint64_t GenAwareKeyword                  = 0x10000000000; // This keyword is necessary for the start and stop events
+        const uint64_t GCHeapAndTypeNamesKeyword        = 0x00001000000; // This keyword is necessary for the type names
+        const uint64_t GCHeapSurvivalAndMovementKeyword = 0x00000400000; // This keyword is necessary for the generation range data.
+        const uint64_t GCHeapDumpKeyword                = 0x00000100000; // This keyword is necessary for enabling walking the heap
+        const uint64_t TypeKeyword                      = 0x00000080000; // This keyword is necessary for enabling BulkType events
+        const uint64_t keyword                          = GenAwareKeyword|GCHeapAndTypeNamesKeyword|GCHeapSurvivalAndMovementKeyword|GCHeapDumpKeyword|TypeKeyword;
+        pProviders[0] = EventPipeProviderConfiguration(W("Microsoft-Windows-DotNETRuntime"), keyword, 5, nullptr);
+        gcGenAnalysisEventPipeSessionId = EventPipe::Enable(
+            outputPath,
+            1024,
+            pProviders,
+            providerCnt,
+            EventPipeSessionType::File,
+            EventPipeSerializationFormat::NetTraceV4,
+            false,
+            nullptr
+        );
+        if (gcGenAnalysisEventPipeSessionId > 0)
+        {
+            gcGenAnalysisEventPipeSession= EventPipe::GetSession(gcGenAnalysisEventPipeSessionId);
+            gcGenAnalysisEventPipeSession->Pause();
+            EventPipe::StartStreaming(gcGenAnalysisEventPipeSessionId);
+            gcGenAnalysisState = GcGenAnalysisState::Enabled;
```

Comment:
I've seen this logic repeated a few places.  Could you wrpa the two halves of this up into a function, so you don't need to update multiple locations if it needs to change?